### PR TITLE
DR-800 Push dependabot pull requests through workflows

### PR DESCRIPTION
**Pull request by timrickwood**
## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .github/workflows/pipeline-deploy-app-develop.yaml [hunk 1]
```diff
@@ -5,9 +5,12 @@ on:
     branches:
       - main
       - 'task/*'
+  pull_request:
+    types: [ labeled ]
 
 jobs:
   metadata:
+    if: ( ${{ github.event.label.name == 'Test Ready' && startswith(github.head_ref, 'dependabot/') }} ) || ( ${{ github.event_name == 'push' && startswith(github.ref_name, 'task/') }} )
```
Comment (timrickwood) [inline .github/workflows/pipeline-deploy-app-develop.yaml:13]: Needs to handle main branch

Comment (timrickwood) [reply .github/workflows/pipeline-deploy-app-develop.yaml:13]: refactor the branch related subclause

File: .github/workflows/test-cleardown.yaml [hunk 2]
```diff
@@ -5,10 +5,11 @@ on:
     branches:
       - main
       - 'task/*'
+      - dependabot/*
 
 jobs:
   metadata:
-    if: github.event.ref_type == 'branch' && startswith(github.event.ref, 'task/') && github.actor != 'github-merge-queue[bot]'
+    if: (github.event.ref_type == 'branch' && startswith(github.event.ref, 'task/') && github.actor != 'github-merge-queue[bot]') || (github.event.ref_type == 'branch' && startswith(github.event.ref, 'dependabot/') && github.actor != 'github-merge-queue[bot]')
```
Comment (timrickwood) [inline .github/workflows/test-cleardown.yaml:12]: Refactor clause

File: .github/workflows/test-cleardown.yaml [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
   delete:
     branches:
-      - main
       - 'task/*'
+      - dependabot/*
 
 jobs:
   metadata:
-    if: github.event.ref_type == 'branch' && startswith(github.event.ref, 'task/') && github.actor != 'github-merge-queue[bot]'
+    if: ( ${{ github.event.ref_type == 'branch' && github.actor != 'github-merge-queue[bot]' }} )
```
Comment (tamaragoldschmidt) [inline .github/workflows/test-cleardown.yaml:16]: Do we no longer care if the branch is task/* or dependabot/* ?

Comment (mmg-nhse) [reply .github/workflows/test-cleardown.yaml:16]: This job will only run on the deletion of task/* & dependabot/* (limes 8 - 12) so we don't need to highlight this again in the if statement.

**Review by tamaragoldschmidt — APPROVED** [review]

**Pull request merged**